corehq/apps/users/views/mobile/users.py

```diff
@@ -730,19 +733,23 @@ def check_username(self, in_data):
             return HttpResponseBadRequest('You must specify a username')
         if username == 'admin' or username == 'demo_user' or username == ANONYMOUS_USERNAME:
             return {'error': _('Username {} is reserved.').format(username)}
-        if '@' in username:
-            return {
-                'error': _('Username {} cannot contain "@".').format(username)
-            }
-        if '&' in username:
-            return {
-                'error': _('Username {} cannot contain "&".').format(username)
-            }
-        if ' ' in username:
+        try:
+            validate_email("{}@example.com".format(username))
+            if BAD_MOBILE_USERNAME_REGEX.search(username) is not None:
+                raise ValidationError("Username contained an invalid character")
```

Comment:
I don't think this message will ever get shown since you catch this error immediately.

Reply:
Yeah, but django forces you to provide some string